Sources/SwiftParser/Nominals.swift

```diff
@@ -304,10 +314,22 @@ extension Parser {
         )
       } while keepGoing != nil && loopProgress.evaluate(currentToken)
     }
+
+    let unexpectedAfterInheritedTypeCollection: RawUnexpectedNodesSyntax?
+
+    // If it contains a left paren, then it might be a python style inheritance clause.
+    // Left paren have been replaced with colon
+    if unexpectedBeforeColon?.containsToken(where: { $0.tokenKind == .leftParen }) == true, let rightParen = self.consume(if: .rightParen) {
```

Comment:
I’m thinking that the code might read easier if we store a boolean variable whether we were in a Python-style inheritance clause instead of inspecting `unexpectedBeforeColon`.

Reply:
Do you think as an extension on the type or?

Reply:
No just a local variable. Like, above you do

```swift
    var isPythonStyleInheritanceClause: Bool
    // Parse python style inheritance clause and replace parentheses with a colon
    if let leftParen = self.consume(if: .leftParen) {
      unexpectedBeforeColon = RawUnexpectedNodesSyntax([leftParen], arena: self.arena)
      colon = missingToken(.colon)
      isPythonStyleInheritanceClause = true
    } else {
      (unexpectedBeforeColon, colon) = self.expect(.colon)
      isPythonStyleInheritanceClause = false
    }
```